app/src/main/java/com/vishaltelangre/nerdcalci/ui/MainActivity.kt

```
@@ -386,52 +499,74 @@ fun CalculatorNavHost(viewModel: CalculatorViewModel, navController: NavHostCont
+            val appVersion = remember {
+                try {
+                    val packageInfo = context.packageManager.getPackageInfo(context.packageName, 0)
+                    val versionName = packageInfo.versionName ?: "Unknown"
+                    val versionCode = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.P) {
+                        packageInfo.longVersionCode
+                    } else {
+                        @Suppress("DEPRECATION")
+                        packageInfo.versionCode.toLong()
+                    }
+                    "v$versionName ($versionCode)"
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)